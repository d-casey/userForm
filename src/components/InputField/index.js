import React from 'react'
import PropTypes from 'prop-types'
import '../../css/input.css'

const InputField = ({ type, label, name, required, register, validation, error, defaultValue, testId }) => {
  return (
    <fieldset className="full-width-imput">
      <legend>
        {label}: {required ? '*' : ''} <span className="error">{error}</span>
      </legend>
      <input type={type} name={name} ref={register(validation)} defaultValue={defaultValue} data-testid={testId} />
    </fieldset>
  )
}

InputField.propTypes = {
  type: PropTypes.string,
  label: PropTypes.string,
  name: PropTypes.string,
  required: PropTypes.bool,
  register: PropTypes.func.isRequired,
  validation: PropTypes.object,
  error: PropTypes.string,
  defaultValue: PropTypes.string
}

export default InputField
