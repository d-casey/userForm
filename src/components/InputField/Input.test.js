import React from 'react'
import { create } from 'react-test-renderer'
import InputField from '.'

test('renders correctly when passed all details', () => {
  const register = jest.fn()
  const component = create(
    <InputField
      type={'text'}
      label={'name'}
      name={'name'}
      required={true}
      register={register}
      validation={{}}
      error={'error'}
      defaultValue={'Severus Snape'}
    />
  )
  expect(component.toJSON()).toMatchSnapshot()
})
