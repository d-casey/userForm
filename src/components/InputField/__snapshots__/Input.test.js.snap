// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when passed all details 1`] = `
<fieldset
  className="full-width-imput"
>
  <legend>
    name
    : 
    *
     
    <span
      className="error"
    >
      error
    </span>
  </legend>
  <input
    defaultValue="Severus Snape"
    name="name"
    type="text"
  />
</fieldset>
`;
